src/scripting_api/doc.js

```diff
@@ -828,15 +828,50 @@ class Doc extends PDFObject {
       return searchedField;
     }
 
+    const parts = cName.split("#");
```

Comment:
It looks like fields should be a tree according to the spec. Why not store the fields as a tree and walk it to find the field?